q2_deblur/plugin_setup.py

```diff
@@ -97,52 +28,89 @@ def denoise(demultiplexed_seqs: SingleLanePerSampleSingleEndFastqDirFmt,
+
+
 plugin.methods.register_function(
-    function=denoise,
+    function=q2_deblur.denoise_16S,
     inputs={
-        'demultiplexed_seqs': SampleData[SequencesWithQuality]
+        'demultiplexed_seqs': SampleData[SequencesWithQuality],
     },
-    parameters={
-        'pos_ref_filepath': qiime2.plugin.Str,
-        'neg_ref_filepath': qiime2.plugin.Str,
-        'mean_error': qiime2.plugin.Float,
-        'indel_prob': qiime2.plugin.Float,
-        'indel_max': qiime2.plugin.Int,
-        'trim_length': qiime2.plugin.Int,
-        'min_reads': qiime2.plugin.Int,
-        'min_size': qiime2.plugin.Int,
-        'negate': qiime2.plugin.Bool,
-        'jobs_to_start': qiime2.plugin.Int,
-        'hashed_feature_ids': qiime2.plugin.Bool
+    parameters=_parameters,
+    outputs=_outputs,
+    input_descriptions={
+        'demultiplexed_seqs': 'The demultiplexed sequences to be denoised.',
     },
-    parameter_descriptions={
-        'pos_ref_filepath': ("Positive filtering database. Keep all sequences "
-                             "aligning to these sequences."),
-        'neg_ref_filepath': ("Negative (artifacts) filtering database. "
-                             "Discard sequences aligning to these sequences."),
-        'mean_error': ("The mean per nucleotide error, used for original "
-                       "sequence estimate. If not passed, a value of 0.5% is "
-                       "used."),
-        'indel_prob': ('Insertion/deletion (indel) probability (same for N '
-                       'indels).'),
-        'indel_max': "Maximum number of insertion/deletions.",
-        'trim_length': "Sequence trim length.",
-        'min_reads': ("Retain only features appearing at least min_reads "
-                      "times across all samples in the resulting feature "
-                      "table."),
-        'min_size': ("In each sample, discard all features with an abundance "
-                     "less than min_size."),
-        'negate': ("Discard (rather than retain) all sequences aligning to "
-                   "the sequences provided in neg_ref_fp."),
-        'jobs_to_start': "Number of jobs to start (if to run in parallel).",
-        'hashed_feature_ids': "If true, hash the feature IDs."
+    parameter_descriptions=_parameter_descriptions,
+    output_descriptions=_output_descriptions,
+    name='Deblur sequences.',
+    description=('Perform sequence quality control using the deblur workflow '
+                 'using a 16S reference. The specific reference used is the '
+                 '88% OTUs from Greengenes 13_8. The reference is used to '
+                 'assess whether each sequence is likely to be 16S by a local '
+                 'alignment using SortMeRNA with a permissive e-value.')
+)
+
+
+plugin.methods.register_function(
+    function=q2_deblur.denoise_other,
+    inputs={
+        'demultiplexed_seqs': SampleData[SequencesWithQuality],
```

Comment:
Deblur does not use the quality data, however, Deblur is also only defined over MiSeq and HiSeq, so quality scores are absolutely feasible here. Unclear to me if we should just require it, as the lack of quality scores increases the potential for the method to be used on non-illumina data.